src/autogenerated/java/net/minestom/server/particle/Particle.java

```diff
@@ -0,0 +1,396 @@
+package net.minestom.server.particle;
+
+import net.kyori.adventure.key.Key;
+import net.kyori.adventure.key.Keyed;
+import net.minestom.server.MinecraftServer;
+import net.minestom.server.color.Color;
+import net.minestom.server.instance.block.Block;
+import net.minestom.server.instance.block.BlockState;
+import net.minestom.server.item.ItemStack;
+import net.minestom.server.particle.data.*;
+import net.minestom.server.particle.data.ParticleEffect;
+import net.minestom.server.registry.Registry;
+import net.minestom.server.utils.NamespaceID;
+import net.minestom.server.utils.binary.BinaryReader;
+import net.minestom.server.utils.clone.PublicCloneable;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * AUTOGENERATED by ParticleTypeGenerator
+ */
+public class Particle implements Keyed, PublicCloneable<Particle> {
+
+    public static final Particle AMBIENT_ENTITY_EFFECT = new Particle(NamespaceID.from("minecraft:ambient_entity_effect"));
```

Comment:
It feels *really* odd having these in the same place as the `Particle` class. Are we sure that this is what we want? There should absolutely be a separation between the types of the particle and the particle data for an individual "instance" of a particle. Not just for separation of concerns, but also because we don't have much control over the auto-generated code, meaning that changes will be really hard to make.

Reply:
I agree, but @TheMode wanted to not have the particle type and its data separated. So I am not really sure how I could make it better without recreating my first approach.